.yarn/sdks/typescript/lib/tsserver.js

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env node
+
+const {existsSync} = require(`fs`);
+const {createRequire} = require(`module`);
+const {resolve} = require(`path`);
+
+const relPnpApiPath = "../../../../.pnp.cjs";
+
+const absPnpApiPath = resolve(__dirname, relPnpApiPath);
+const absRequire = createRequire(absPnpApiPath);
+
+const moduleWrapper = tsserver => {
+  if (!process.versions.pnp) {
+    return tsserver;
+  }
+
+  const {isAbsolute} = require(`path`);
+  const pnpApi = require(`pnpapi`);
+
+  const isVirtual = str => str.match(/\/(\$\$virtual|__virtual__)\//);
+  const isPortal = str => str.startsWith("portal:/");
+  const normalize = str => str.replace(/\\/g, `/`).replace(/^\/?/, `/`);
+
+  const dependencyTreeRoots = new Set(pnpApi.getDependencyTreeRoots().map(locator => {
+    return `${locator.name}@${locator.reference}`;
+  }));
+
+  // VSCode sends the zip paths to TS using the "zip://" prefix, that TS
```

Comment:
I VSCode configuration necessary for using typescript? 

And did all this code from someplace?

Reply:
I think this was necessary to make VS Code work with Yarn's PnP. I think without this, VS Code can't find type definitions.

All this was generated by Yarn from running `yarn dlx @yarnpkg/sdks`.
https://yarnpkg.com/getting-started/editor-sdks

Reply:
I just read the PR description, so now I know where a lot of this came from. :) But I'm still curious about VSCode.

Reply:
I suppose the VS Code config / SDK could be ignored, but then everyone who uses VS Code would have to install the SDK themselves.

Reply:
And people use VSCode? If so, I agree with keeping it in.

Reply:
I use it. I don't know how many others (if any) do.